crates/proto/src/op/message.rs

```diff
@@ -893,27 +887,25 @@ pub fn update_header_counts(
 /// Alias for a function verifying if a message is properly signed
 pub type MessageVerifier = Box<dyn FnMut(&[u8]) -> ProtoResult<DnsResponse> + Send>;
 
-/// A trait for performing final amendments to a Message before it is sent.
-///
-/// An example of this is a SIG0 signer, which needs the final form of the message,
-///  but then needs to attach additional data to the body of the message.
+/// A trait for adding a final `MessageSignature` to a Message before it is sent.
 pub trait MessageFinalizer: Send + Sync + 'static {
-    /// The message taken in should be processed and then return [`Record`]s which should be
-    ///  appended to the additional section of the message.
+    /// Finalize the provided `Message`, computing a `MessageSignature`, and optionally
```

Comment:
Should we also rename the trait and its method, like `MessageSigner::sign_message()`?

Reply:
I also did a quick scan through [GitHub code search](https://github.com/search?type=code&q=%22impl+MessageFinalizer%22+language%3ARust&l=Rust) and as expected, found no implementations of the trait outside of the ones provided in-crate.

Reply:
Tacked on a final rename commit.